core/chains/evm/config/v2/config.go

```diff
@@ -0,0 +1,274 @@
+	}
+	if cfg.EvmGasBumpPercent.Valid {
+		v := uint16(cfg.EvmGasBumpPercent.Int64)
+		c.GasBumpPercent = &v
+	}
+	if cfg.EvmGasBumpTxDepth.Valid {
+		v := uint16(cfg.EvmGasBumpTxDepth.Int64)
+		c.GasBumpTxDepth = &v
+	}
+	c.GasBumpWei = cfg.EvmGasBumpWei
+	c.GasFeeCapDefault = cfg.EvmGasFeeCapDefault
+	if cfg.EvmGasLimitDefault.Valid {
+		c.GasLimitDefault = utils.NewBigI(cfg.EvmGasLimitDefault.Int64)
+	}
+	if cfg.EvmGasLimitMultiplier.Valid {
+		v := decimal.NewFromFloat(cfg.EvmGasLimitMultiplier.Float64)
+		c.GasLimitMultiplier = &v
+	}
+	c.GasPriceDefault = cfg.EvmGasPriceDefault
+	c.GasTipCapDefault = cfg.EvmGasTipCapDefault
+	c.GasTipCapMinimum = cfg.EvmGasTipCapMinimum
+	if cfg.EvmHeadTrackerHistoryDepth.Valid {
+		if c.HeadTracker == nil {
+			c.HeadTracker = &HeadTracker{}
+		}
+		v := uint32(cfg.EvmHeadTrackerHistoryDepth.Int64)
+		c.HeadTracker.HistoryDepth = &v
+	}
+	if cfg.EvmHeadTrackerMaxBufferSize.Valid {
+		if c.HeadTracker == nil {
+			c.HeadTracker = &HeadTracker{}
+		}
+		v := uint32(cfg.EvmHeadTrackerMaxBufferSize.Int64)
+		c.HeadTracker.MaxBufferSize = &v
+	}
+	if i := cfg.EvmHeadTrackerSamplingInterval; i != nil {
+		if c.HeadTracker == nil {
+			c.HeadTracker = &HeadTracker{}
+		}
+		c.HeadTracker.SamplingInterval = cfg.EvmHeadTrackerSamplingInterval
+	}
+	if cfg.EvmLogBackfillBatchSize.Valid {
+		v := uint32(cfg.EvmLogBackfillBatchSize.Int64)
+		c.LogBackfillBatchSize = &v
+	}
+	c.LogPollInterval = cfg.EvmLogPollInterval
+	c.MaxGasPriceWei = cfg.EvmMaxGasPriceWei
+	if cfg.EvmNonceAutoSync.Valid {
+		c.NonceAutoSync = &cfg.EvmNonceAutoSync.Bool
+	}
+	if cfg.EvmUseForwarders.Valid {
+		c.UseForwarders = &cfg.EvmUseForwarders.Bool
+	}
+	if cfg.EvmRPCDefaultBatchSize.Valid {
+		v := uint32(cfg.EvmRPCDefaultBatchSize.Int64)
+		c.RPCDefaultBatchSize = &v
+	}
+	if cfg.FlagsContractAddress.Valid {
+		s := cfg.FlagsContractAddress.String
+		if !common.IsHexAddress(s) {
+			return fmt.Errorf("invalid FlagsContractAddress: %s", s)
+		}
+		a := common.HexToAddress(s)
+		v := ethkey.EIP55AddressFromAddress(a)
+		c.FlagsContractAddress = &v
+	}
+	if cfg.GasEstimatorMode.Valid {
+		c.GasEstimatorMode = &cfg.GasEstimatorMode.String
+	}
+	for s, kcfg := range cfg.KeySpecific {
+		if !common.IsHexAddress(s) {
+			return fmt.Errorf("invalid address KeySpecific: %s", s)
+		}
+		a := common.HexToAddress(s)
+		v := ethkey.EIP55AddressFromAddress(a)
+		c.KeySpecific = append(c.KeySpecific, KeySpecific{
+			Key:            &v,
+			MaxGasPriceWei: kcfg.EvmMaxGasPriceWei,
+		})
+	}
+	if cfg.LinkContractAddress.Valid {
+		s := cfg.LinkContractAddress.String
+		if !common.IsHexAddress(s) {
+			return fmt.Errorf("invalid LinkContractAddress: %s", s)
+		}
+		a := common.HexToAddress(s)
+		v := ethkey.EIP55AddressFromAddress(a)
+		c.LinkContractAddress = &v
+	}
+	if cfg.OperatorFactoryAddress.Valid {
+		s := cfg.OperatorFactoryAddress.String
+		if !common.IsHexAddress(s) {
+			return fmt.Errorf("invalid OperatorFactoryAddress: %s", s)
```

Comment:
can we pls use errors.Errorf everywhere? It's just nice to have the stacktrace if its logged

Reply:
Agreed.